src/framework/schema/schema.ts

```diff
@@ -13,26 +13,15 @@ export type SettingsInput = {
   /**
    * todo
    */
-  connections?: ConnectionConfig & {
-    // We tried the idea of types.default: false | ConnectionConfig
-    // but got blocked by https://github.com/microsoft/TypeScript/issues/17867
-
-    /**
-     * todo
-     *
-     * @default `false`
-     */
-    disableDefaultType?: boolean
+  connections?: {
     /**
      * todo
      */
-    types?: {
-      default?: ConnectionConfig
-      // Extra undefined below is forced by it being above, forced via `?:`.
-      // This is a TS limitation, cannot express void vs missing semantics,
-      // being tracked here: https://github.com/microsoft/TypeScript/issues/13195
-      [typeName: string]: ConnectionConfig | undefined
-    }
+    default?: ConnectionConfig | false
+    // Extra undefined below is forced by it being above, forced via `?:`.
+    // This is a TS limitation, cannot express void vs missing semantics,
+    // being tracked here: https://github.com/microsoft/TypeScript/issues/13195
+    [typeName: string]: ConnectionConfig | undefined | false
```

Comment:
So our solution to https://github.com/microsoft/TypeScript/issues/17867 is to allow `false` for all cases? I guess you feel this is less evil than needing to do the nesting and have `disableDefaultType` at the top level?

The tradeoff (until TS solves their side 🤞🏻) is that now we get a meaningless `false` option in the API for non-default connection types. Or, do you see value here somehow?

If we're agreed that ideally the `false` union member goes away eventually, then retain the comment (LOC 17-18) as a TODO for future reference.

Reply:
> So our solution to microsoft/TypeScript#17867 is to allow false for all cases? I guess you feel this is less evil than needing to do the nesting and have disableDefaultType at the top level?

Yes. `false` can actually still be useful-ish in case someone wants to be "explicit" about disabling a previously enabled custom type instead of completely removing it.

I agree that I probably wouldn't have put `false` for the custom types if I could, but I def. still find that better that before. Although `false` might no be useful in most cases, it's, at worst, just more consistent with the `default` API which accepts `false`

Reply:
Agreed its a great tradeoff! Why couldn't I come to that conclusion alone 😆 collab FTW 🙌